### PR TITLE
Issue #14899: add latest version of javaguide.html

### PR DESCRIPTION
**Pull request by Zopsss**
Issue #14899 

Add the latest version of [javaguide.html](https://google.github.io/styleguide/javaguide.html). The html file and other required files were copied from https://github.com/google/styleguide

### CODE REVIEW

**Comment by romani** [comment]
Git history of style guide https://github.com/google/styleguide/commits/gh-pages/javaguide.html

Last update 20220202.

**Comment by romani** [comment]
GitHub, generate website

**Comment by romani** [comment]
We need to update https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/2506b72_2024144131/google_style.html#Google_Useful_information section.
To mention new version of cached page and mention that until September 2024, coverage will in transition from previous cashed to new cached.

**Comment by romani** [comment]
We will continue to have releases each end of month. So we need to figure out a way to show work in progress, as each release is independent and ready to use by users, doc should be self explained.

**Comment by rnveach** [comment]
@romani You didn't mention links in 2nd column at https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/2506b72_2024144131/google_style.html#a1 . They all point to 2018 document. When does those get updated if not in this PR?

**Comment by romani** [comment]
Final update will be at the end of June.
During this month we need to update all links to point to new style guide and update all tables rows to mention issue numbers if something needs to be improved to get coverage as described in new style.

So it is ok that green icons will be switched to blue with links to issues after fixing which coverage will improve. 
So in this case our style coverage will be actual release by release.

**Comment by Zopsss** [comment]
> We will continue to have releases each end of month. So we need to figure out a way to show work in progress, as each release is independent and ready to use by users, doc should be self explained.

We can add a new paragraph in the `ATTENTION` section mentioning that the style guide is currently being updated based on the latest version, we can mention the section too till which the implementation has done. 

We can also make a copy of the current `google_checks.xml` and provide a link of it in the newly added paragraph if some user want to use the old version instead of a mix version.

Let me know if I should proceed with this or not @romani

**Comment by Zopsss** [comment]
> Does GitHub render it for you as 20220203 ? Just curious.

Yes for me it is showing Feb 3, 2022

**Comment by Zopsss** [comment]
> During this month we need to update all links to point to new style guide 

So we will be updating the links section by section as we progress with the implementation right?

**Comment by rnveach** [comment]
> Last update 20220202.
Does GitHub render it for you as 20220203 ? Just curious.

> Yes for me it is showing Feb 3, 2022

It is probably just a timezone difference.

**Comment by Zopsss** [comment]
> We need to update https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/2506b72_2024144131/google_style.html#Google_Useful_information section. To mention new version of cached page and mention that until September 2024, coverage will in transition from previous cashed to new cached.

done. 

https://github.com/checkstyle/checkstyle/blob/f07c4239691f830b3c709a957bf1c89f565bd20d/src/xdocs/google_style.xml#L15-L22

File: src/xdocs/google_style.xml [hunk 1]
```diff
@@ -16,9 +16,9 @@
           This coverage report was created for
           <a href="https://google.github.io/styleguide/javaguide.html">
             Google Java Style</a>(
-          <a href="styleguides/google-java-style-20180523/javaguide.html">
+          <a href="styleguides/google-java-style-20220203/javaguide.html">
             cached page</a>),
-          version of 23 May 2018, current as of 07 May 2019
+          version of 3 Feb 2022, current as of September 2024
```
Comment (rnveach) [inline src/xdocs/google_style.xml:21]: Why September? Shouldn't this be today's date as this is when we are capture this stuff.

Comment (rnveach) [reply src/xdocs/google_style.xml:21]: > that until September 2024, coverage will in transition from previous cashed to new cached

I don't think this is what @romani was mentioning. There is no mention there will be a transition period.

@romani please confirm

Comment (Zopsss) [reply src/xdocs/google_style.xml:21]: yeah I also thought same of why September, but @romani mentioned September here, so I just followed his suggestions:

> We need to update https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/2506b72_2024144131/google_style.html#Google_Useful_information section.
To mention new version of cached page and mention that until September 2024

Comment (romani) [reply src/xdocs/google_style.xml:21]: We have cached page of 2018, let's keep it for now.
We add the new more cached page, that is not yet active.
So coverage will be split for few months, or I hope we will migrate to new cached style until end on June.

In few weeks we should have only one cached style page, all links reference to new cached page and coverage table is referencing issues that will cover gaps in coverage.
In this case we can release at any point of time and we always be correct.

So let's make wording something like:
This coverage report was created for Google Java Style( cached page), version of 07 May 2019, new target of coverage is Google Java Style( cached page), version of 03 Feb 2022.

Comment (romani) [reply src/xdocs/google_style.xml:21]: As we finish review and issues creation at https://github.com/checkstyle/checkstyle/pull/14901 , we will remove old cached file and all will be referencing new cached version and all new issues to improve will be in table.
If issues will be a lot, we will reference umbrella issue only.

Comment (Zopsss) [reply src/xdocs/google_style.xml]: Done, I have little bit modified from what you suggested:

https://github.com/checkstyle/checkstyle/blob/dbc233cb8dbf8db594a0392ead85f9fe9d06b33d/src/xdocs/google_style.xml#L15-L27

**Comment by Zopsss** [comment]
@romani can you merge this PR? So I can reference the new version of cached page in my upcoming PRs

**Comment by romani** [comment]
Github, generate website

**Review by romani — APPROVED** [review]
ok to merge is renders welll

**Pull request merged**